platform/security/common/src/main/java/org/codice/ddf/security/common/Security.java

```diff
@@ -288,14 +315,42 @@ public Date getExpires(Subject subject) {
    * @return reference to the {@link SecurityManager}
    */
   public SecurityManager getSecurityManager() {
-    BundleContext context = getBundleContext();
-    if (context != null) {
-      ServiceReference securityManagerRef = context.getServiceReference(SecurityManager.class);
-      return (SecurityManager) context.getService(securityManagerRef);
```

Comment:
This issue was that the SecurityManager service was not available by the time that the platform-scheduler needed it. The `securityManagerRef` was null, and line 294 would throw a `java.lang.NullPointerException: A null service reference is not allowed.` error.

After discussing solutions with @stustison, we decided to add retry logic to this Security class, as opposed to the platform-scheduler,  when trying to get the SecurityManager so that it will not be an issue elsewhere in DDF.